estate/models/estate_property.py

```diff
@@ -0,0 +1,108 @@
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+
+class EstateProperty(models.Model):
+    _name = 'estate.property'
+    _description = 'Create real estate properties and keep track of status'
+    _order = 'id desc'
+
+    name = fields.Char(string="Name", required=True)
+    description = fields.Text(string="Description")
+    postcode = fields.Char(string="Postcode")
+    date_availability = fields.Date(string="Date Availability", default=lambda self: fields.Datetime.today() + relativedelta(months=3), copy=False)
+    expected_price = fields.Float(string="Expected Price", required=True)
+    selling_price = fields.Float(string="Selling Price", readonly=True, copy=False)
+    bedrooms = fields.Integer(string="Bedrooms", default=2)
+    living_area = fields.Integer(string="Living Area")
+    facades = fields.Integer(string="Facades")
+    garage = fields.Boolean(string="Garage")
+    garden = fields.Boolean(string="Garden")
+    garden_area = fields.Integer(string="Garden Area")
+    garden_orientation = fields.Selection(
+        string="Garden Orientation",
+        selection=[('north', "North"), ('south', "South"), ('east', "East"), ('west', "West")]
+    )
+    active = fields.Boolean(string="Active", default=True)
+    state = fields.Selection(
+        string="State",
+        default='new',
+        selection=[
+            ('new', "New"),
+            ('offer_received', "Offer Received"),
+            ('offer_accepted', "Offer Accepted"),
+            ('sold', "Sold"),
+            ('cancelled', "Cancelled")
+        ]
+    )
+    property_type_id = fields.Many2one('estate.property.types', string="Property Type")
+    salesman_id = fields.Many2one('res.users', string="Salesman", default=lambda self: self.env.user)
+    buyer_id = fields.Many2one('res.partner', string="Buyer")
+    tag_ids = fields.Many2many('estate.property.tag', string="Property Tags")
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string="Offers")
+    total_area = fields.Float(compute="_compute_total_area", string="Total Area")
+    best_price = fields.Float(compute="_compute_best_price", string="Best Price")
+
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price > 0)',
+         "The expected price should be greater than 0."),
+        ('check_selling_price', 'CHECK(buyer_id IS NULL OR selling_price > 0)',
+         "The selling price should be greater than 0."),
+    ]
+
+    @api.constrains('selling_price', 'expected_price')
+    def _check_prices_difference(self):
+        for record in self:
+            if record.buyer_id and float_compare(record.selling_price, 0.9 * record.expected_price, 5) == -1:
```

Comment:
Use of a meaningful variable name to ease the understanding of the method
```suggestion
        for estate in self:
            if estate.buyer_id and float_compare(record.selling_price, 0.9 * record.expected_price, 5) == -1:
```